server/ot_rpc.go

```diff
@@ -73,23 +77,40 @@ func runTranscoder(n *core.LivepeerNode, orchAddr string, capacity int) error {
 		}
 		return err
 	}
+	exitc := make(chan os.Signal)
+	signal.Notify(exitc, os.Interrupt)
+	defer signal.Stop(exitc)
+	go func() {
+		select {
+		case sig := <-exitc:
+			glog.Infof("Exiting Livepeer Transcoder: %v", sig)
+			// Cancelling context will close connection to orchestrator
+			cancel()
+			return
+		}
+	}()
 
 	httpc := &http.Client{Transport: &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}}
+	var wg sync.WaitGroup
 	for {
 		notify, err := r.Recv()
 		if err != nil {
 			status := status.Convert(err)
 			if status.Message() == SecretErr.Error() { // consider this unrecoverable
 				return SecretErr
 			}
+			glog.Infof("End of stream recieve cylcle because err=%v, waiting for running transcode jobs to complete", err)
```

Comment:
Would it make sense to check for the `codes.Canceled` here and return a `TranscoderShutdownErr` to get a more readable message in `RunTranscoder`?

Then it might also make sense to adapt the error-wrapping approach that O uses to signal fatal/non-fatal errors so we only need to check the error type once in `RunTranscoder`?

Reply:
Fixed